extensions/ipynb/src/helper.ts

```diff
@@ -0,0 +1,139 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export function deepClone<T>(obj: T): T {
```

Comment:
we don't have more generally available versions of these functions? (and better tested?)

Reply:
@amunger currently we don't, I borrowed this from the core.